### PR TITLE
Updating ClusterPool scale modal

### PR DESCRIPTION
**Pull request by ellajuengst**
https://github.com/stolostron/backlog/issues/19598
Users can scale the running count property of a cluster pool
Signed-off-by: Ella Juengst <ellajuengst@gmail.com>

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -356,12 +356,13 @@
     "clusterPool.creation.success.message": "{{name}} was successfully created.",
     "clusterPool.creation.success.title": "Cluster pool created",
     "clusterPool.destroy": "Destroy cluster pool",
-    "clusterPool.modal.scale.input": "Set desired cluster pool size",
-    "clusterPool.modal.scale.message": "Specify the desired size of <bold>{{clusterPoolName}}</bold>. Adjusting the size of the cluster pool will result in the creation or destruction of clusters; only unclaimed clusters will be destroyed if downsizing.",
+    "clusterPool.modal.scale.message": "Specify the desired size of <bold>{{clusterPoolName}}</bold>. Adjusting the size or running count of the cluster pool will result in the creation or destruction of clusters; only unclaimed clusters will be destroyed if downsizing.",
     "clusterPool.modal.scale.poolName": "Cluster pool name",
     "clusterPool.modal.scale.poolNamespace": "Cluster pool namespace",
+    "clusterPool.modal.scale.runningCount.input": "Set desired number of running clusters for this cluster pool.",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:362]:
```suggestion
    "clusterPool.modal.scale.runningCount.input": "Set desired number of running clusters",
```

Remove the period, and "for this cluster pool" seems unnecessary.

File: frontend/public/locales/en/translation.json [hunk 2]
```diff
@@ -356,12 +356,13 @@
...
+    "clusterPool.modal.scale.size.input": "Set desired cluster pool size",
     "clusterPool.modal.scale.title": "Scale cluster pool",
-    "clusterPool.modal.scale.validation.greaterThanZero": "clusterPool.modal.scale.validation.greaterThanZero",
+    "clusterPool.modal.scale.validation.greaterThanZero": "The cluster pool size and running count must both be greater than 0.",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:365]: Your validation checks for less than 0. Equal to 0 is valid for both pool size and running count.

File: frontend/src/routes/Infrastructure/Clusters/ClusterPools/components/ScaleClusterPoolModal.tsx [hunk 3]
```diff
@@ -31,9 +32,18 @@ export function ScaleClusterPoolModal(props: ScaleClusterPoolModalProps) {
...
     }, [props.clusterPool])
 
+    useEffect(() => {
+        if (props.clusterPool && !!props.clusterPool.spec?.runningCount) {
+            setRunningCount(props.clusterPool.spec?.runningCount!)
+        } else if (!props.clusterPool) {
+            setRunningCount(0)
+        }
+    }, [props.clusterPool])
+
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Clusters/ClusterPools/components/ScaleClusterPoolModal.tsx:42]: You could combine these into a single useEffect.

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
A few minor changes requested, please.

File: frontend/public/locales/en/translation.json [hunk 4]
```diff
@@ -363,12 +363,13 @@
     "clusterPool.creation.success.message": "{{name}} was successfully created.",
     "clusterPool.creation.success.title": "Cluster pool created",
     "clusterPool.destroy": "Destroy cluster pool",
-    "clusterPool.modal.scale.input": "Set desired cluster pool size",
-    "clusterPool.modal.scale.message": "Specify the desired size of <bold>{{clusterPoolName}}</bold>. Adjusting the size of the cluster pool will result in the creation or destruction of clusters; only unclaimed clusters will be destroyed if downsizing.",
+    "clusterPool.modal.scale.message": "Specify the desired size of <bold>{{clusterPoolName}}</bold>. Adjusting the size or running count of the cluster pool will result in the creation or destruction of clusters; only unclaimed clusters will be destroyed if downsizing.",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:366]: I'm not sure if it is accurate to say that adjusting the running count will result in the creation or destruction of clusters. Usually it should just affect how many are running vs. hibernating. I think we confirmed that if `running` is greater than `size`, you just get `size` running clusters. And when reducing `runningCount` I would think that running cluster are then just hibernated. Please check and then clarify.

Comment (ellajuengst) [reply frontend/public/locales/en/translation.json]: This is correct, just fixed the message

**Pull request merged**